Index out of range · artifacts.(*ArtifactServer).GetOutputArtifact

https://oss-fuzz.com/testcase-detail/5254033034575872

```


panic: runtime error: index out of range [2] with length 2
--
  |  
  | goroutine 17 [running, locked to thread]:
  | github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).getArtifact(0x10c0002b9428, {0x2b322c0, 0x10c0002c1760}, 0x10c0003abd00, 0x0)
  | github.com/argoproj/argo-workflows/v3/server/artifacts/artifact_server.go:58 +0x5f8
  | github.com/argoproj/argo-workflows/v3/server/artifacts.(*ArtifactServer).GetOutputArtifact(...)
  | github.com/argoproj/argo-workflows/v3/server/artifacts/artifact_server.go:49
```

--- a/server/artifacts/artifact_server.go
+++ b/server/artifacts/artifact_server.go
@@ -2,6 +2,7 @@ package artifacts
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
@@ -55,6 +56,10 @@ func (a *ArtifactServer) GetInputArtifact(w http.ResponseWriter, r *http.Request
 
 func (a *ArtifactServer) getArtifact(w http.ResponseWriter, r *http.Request, isInput bool) {
 	requestPath := strings.SplitN(r.URL.Path, "/", 6)
+	if len(requestPath) != 6 {
+		a.serverInternalError(errors.New("request path is not valid"), w)
+		return
+	}
 	namespace := requestPath[2]
 	workflowName := requestPath[3]
 	nodeId := requestPath[4]

